### ISSUE TITLE
Crash from invalid inventory when contraption disassembles 

### ISSUE THREAD

**Issue by StevenG5** [Fabricators-of-Create/Create#1038]
### Describe the Bug

I run a small server with about 200 players. One of my players discovered today he could instigate a crash but moving/jumping while on a Gantry system.

### Reproduction Steps

1. Create a Contraption that uses a Gantry System
2. Start the Gantry System
3. Jump on the contraption while the Gantry System is moving
4. Crash
...

### Expected Result

Did not expect to crash and was expected to be allowed to move freely on the gantry system.

### Crash Report or Log

https://pastebin.com/tz13Ygu4

### Operating System

Windows 10

### Mod Version

0.5.1b

### Minecraft Version

1.19.2

### Other Mods

We use the "Create Optimized" Modpack on curseforge, but judging by the logs and my player's report, this is an isolated issue.

**Comment by TropheusJ** [comment]
this has nothing to do with jumping or gantries, an inventory is in an invalid state when the contraption disassembles. Can you get it to happen again? I need a reliable setup.

**Comment by StevenG5** [comment]
**We have banned the gantry shaft from being crafted, placed, or created and it has solved the problem.** We run the largest public server for the modpack so I'm not super keen on lifting the ban to test it out again since there are always players online. 

I will try it in singleplayer and see if it can be replicated but I have 5 separate crash reports for our server; all of which state the coordinates of a gantry shaft contraption. My players also all told me they'd moved/jumped on the contraption to cause the server to crash (which is why I mentioned it in the report). , We have **thousands** of contraptions and **hundreds** of unique player bases; **the only contraptions to cause crashes are contraptions with the gantry shaft.** Regardless, I have attached all 5 crash files to this message in hopes you'll be able to piece together some more information.

[crash-2023-08-05_19.40.31-server.txt](https://github.com/Fabricators-of-Create/Create/files/12312831/crash-2023-08-05_19.40.31-server.txt)
[crash-2023-08-05_19.08.59-server.txt](https://github.com/Fabricators-of-Create/Create/files/12312832/crash-2023-08-05_19.08.59-server.txt)
[crash-2023-08-05_10.12.20-server (1).txt](https://github.com/Fabricators-of-Create/Create/files/12312833/crash-2023-08-05_10.12.20-server.1.txt)
[crash-2023-08-05_09.57.40-server.txt](https://github.com/Fabricators-of-Create/Create/files/12312834/crash-2023-08-05_09.57.40-server.txt)
[crash-2023-08-05_09.41.08-server.txt](https://github.com/Fabricators-of-Create/Create/files/12312835/crash-2023-08-05_09.41.08-server.txt)

**Comment by DonorKrovi** [comment]
The error message indicates that an IllegalArgumentException has been thrown with the message "Transfer variant may not be blank", which means "The transfer option cannot be blank". This may mean that when you try to transfer an item to storage, the option or parameter that defines this transfer was not correctly set or is missing.

**Comment by TropheusJ** [comment]
Thank you ChatGPT

**Comment by RobusTetus** [comment]
Me and few of my friends started a server recently with create. We managed to hit I think same issue. However with different blocks. First three crashes were caused by `create:cart_assembler` the last and most recent one caused by `create:mechanical_bearing`.

This makes me think it is not tied directly to the `gantry_shaft` item discussed above and not limited to the `1.19.2` version since we are experiencing it on the `1.20.1` version.

[crash-2024-02-25_18.01.06-server.txt](https://github.com/Fabricators-of-Create/Create/files/14397141/crash-2024-02-25_18.01.06-server.txt)
[crash-2024-02-24_14.42.47-server.txt](https://github.com/Fabricators-of-Create/Create/files/14397142/crash-2024-02-24_14.42.47-server.txt)
[crash-2024-02-24_13.11.02-server.txt](https://github.com/Fabricators-of-Create/Create/files/14397143/crash-2024-02-24_13.11.02-server.txt)
[crash-2024-02-23_17.21.25-server.txt](https://github.com/Fabricators-of-Create/Create/files/14397144/crash-2024-02-23_17.21.25-server.txt)

**Comment by TropheusJ** [comment]
have you been taking items out of assembled contraption inventories? Moving Chest / Moving Barrel etc

**Comment by RobusTetus** [comment]
> have you been taking items out of assembled contraption inventories? Moving Chest / Moving Barrel etc

Now that you mention it. It crashed only once when I was on a server. I was working with a cart assembler which caused the crash in the end. I destroyed maybe two barrels full of items accidentally belonging to the contraption when I was rebuilding it. The crash occurred once I flipped a lever next to the cart assembler to raise the contraption up. I don't recall the exact procedure which led to the crash, but it was something in that sense.
